src/test/java/com/scottlogic/TopicFinderTest.java

```diff
@@ -0,0 +1,106 @@
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.time.OffsetDateTime;
+import java.time.ZoneOffset;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Stream;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+
+public class TopicFinderTest {
+
+
+    private static List<String> stopWords;
+    private static TopicFinder topicFind;
+
+    private UserPost createUserPost(String content) {
+        return new UserPost("Joe Bloggs",
+                OffsetDateTime.of(2020, 1, 3, 7, 12, 3, 0, ZoneOffset.UTC),
+                content, 2);
+    }
+
+    @BeforeClass
+    public static void loadStopWords() throws IOException {
+        File file = new File("C:/Dev/post-sorting-grad-project/src/main/resources/English_StopWords.txt");
+        stopWords = Files.readAllLines(Paths.get(String.valueOf(file)));
+        topicFind = new TopicFinder(stopWords);
+    }
+
+
+
+    @ParameterizedTest
+    @ValueSource(strings = {"An example of a post \nwith line breaks.", "An! example, of a post with line breaks","An! example, of a- post with line breaks"})
+    public void topicFinder_generalPostParameterized_stopWordsPunctuationRemoved(String content) throws IOException {
+
+        File file = new File("C:/Dev/post-sorting-grad-project/src/main/resources/English_StopWords.txt");
+        stopWords = Files.readAllLines(Paths.get(String.valueOf(file)));
+        TopicFinder topicFind = new TopicFinder(stopWords);
+        UserPost userPost = createUserPost(content);
+
+        ArrayList<String> actualResult = topicFind.findTopics(userPost);
+        ArrayList<String> expectedResult = new ArrayList<String>(Arrays.asList("example", "post", "line", "breaks"));
+
+        assertEquals(expectedResult, actualResult);
+    }
+
+    @ParameterizedTest
+    @MethodSource("provideContentForParameterizedTests")
+    public void topicFinder_parametersThroughMethod_stopWordsPunctuationRemoved(String content) throws IOException {
+        File file = new File("C:/Dev/post-sorting-grad-project/src/main/resources/English_StopWords.txt");
+        stopWords = Files.readAllLines(Paths.get(String.valueOf(file)));
+        TopicFinder topicFind = new TopicFinder(stopWords);
+        UserPost userPost = createUserPost(content);
+
+        ArrayList<String> actualResult = topicFind.findTopics(userPost);
+        ArrayList<String> expectedResult = new ArrayList<String>(Arrays.asList("example", "post", "line", "breaks"));
+
+        assertEquals(expectedResult, actualResult);
+    }
+    private static Stream<Arguments> provideContentForParameterizedTests() {
+        return Stream.of(
+          Arguments.of("An example of a post \nwith line breaks."),
+                Arguments.of("An! example, of a post with line breaks"),
+                Arguments.of("An! example, of a- post with line breaks")
+        );
+    }
```

Comment:
This could be simplified to return a stream of strings, without having to make an Arguments object from each one.